### PR TITLE
Dev/trash

### PR DESCRIPTION
**Pull request by arjund-06**
### Functionality:
Show the tasks that have been deleted.

### Solution:
Made the deleted tasks work like BackLog and any other sprint with the Main details like TotalTasks etc..

### Risk level:
- [x] high 
- [ ] medium
- [ ] low

### How to test:
1. firebase emulators:start
2. ng serve
3. Create a new sprint and new task.
4. Now delete that task
5. In board component click the Deleted tab in focus nav and view deleted tasks

### Pics:

image

image

### CODE REVIEW

File: functions/deleteTaskAPI.js [hunk 1]
```diff
@@ -72,7 +71,76 @@ exports.deleteTask = functions.https.onRequest((request, response) => {
...
+                }
+
+
+                totalNumberOfTask = totalNumberOfTask + 1;
+                totalUnCompletedTask = totalUnCompletedTask + 1;
+
+                sprintDataPromise = db.collection("Main").doc("Deleted").set({
+                    EndDate: "xx/xx/xxxx",
+                    StartDate: "xx/xx/xxxx",
+                    Status: "Not Started",
```
Comment (shivamgoyal7) [inline functions/deleteTaskAPI.js:130]: these things are not required here.
You can disable sprint details

File: functions/deleteTaskAPI.js [hunk 2]
```diff
@@ -72,7 +71,76 @@ exports.deleteTask = functions.https.onRequest((request, response) => {
...
+                    totalDevelopmentTask = totalDevelopmentTask + 1;
+                } else if (category === "Business") {
+                    totalBusinessTask = totalBusinessTask + 1;
+                } else if (category === "Marketing") {
+                    totalMarketingTask = totalMarketingTask + 1;
+                } else {
+                    totalOtherTask = totalOtherTask + 1;
+                }
+
+
```
Comment (shivamgoyal7) [inline functions/deleteTaskAPI.js:123]: remove this extra line no. 123

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]
you can disable sprint details if possible..
or
instead of disabling put "---" in the place of xx/xx/xx and not started

**Comment by shivamgoyal7** [comment]
resolve conflict

File: functions/deleteTaskAPI.js [hunk 3]
```diff
@@ -72,7 +71,75 @@ exports.deleteTask = functions.https.onRequest((request, response) => {
...
-        const deleteTaskPromises = [p1, p2];
+        const p3 = db.collection("Main").doc("Trash").get().then((doc) => {
+            if (doc.exists) {
+                totalNumberOfTask = doc.data().TotalNumberOfTask;
+                totalDevelopmentTask = doc.data().TotalDevelopmentTask;
+                totalBusinessTask = doc.data().TotalBusinessTask;
+                totalMarketingTask = doc.data().TotalMarketingTask;
+                totalOtherTask = doc.data().TotalOtherTask;
+                totalCompletedTask = doc.data().TotalCompletedTask;
+                totalUnCompletedTask = doc.data().TotalUnCompletedTask;
```
Comment (shivamgoyal7) [inline functions/deleteTaskAPI.js:82]: we dont need these counter totalCompletedTask, totalUnCompletedTask
and also change others name to totalDevelopmentTaskDeleted....

File: functions/deleteTaskAPI.js [hunk 4]
```diff
@@ -72,7 +71,75 @@ exports.deleteTask = functions.https.onRequest((request, response) => {
...
+                }
+                totalNumberOfTask = totalNumberOfTask + 1;
+                const updateDeleteTaskCounter2 = db.collection("Main").doc("Trash").update({
+                    TotalDevelopmentTask: totalDevelopmentTask,
+                    TotalBusinessTask: totalBusinessTask,
+                    TotalMarketingTask: totalMarketingTask,
+                    TotalOtherTask: totalOtherTask,
+                    TotalNumberOfTask: totalNumberOfTask,
+                    TotalCompletedTask: totalCompletedTask,
+                    TotalUnCompletedTask: totalUnCompletedTask,
```
Comment (shivamgoyal7) [inline functions/deleteTaskAPI.js:101]: delete this

**Review by shivamgoyal7 — CHANGES_REQUESTED** [review]

**Comment by arjund-06** [comment]
I am closing this Pull request cuz I made all necessary changes in #164 with the updated API.

**Pull request closed without merging**